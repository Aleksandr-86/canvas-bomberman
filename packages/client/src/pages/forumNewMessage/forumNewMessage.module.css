.forum {
  width: 100vw;
  height: 100vh;
  background-image: url('../../assets/images/forumBackground.png');
  background-size: cover;
  background-repeat: no-repeat;
}

.forumWrapper {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  width: 100vw;
  height: 100vh;
  background: hsl(120deg 1% 45% / 75%);
  backdrop-filter: blur(0.5vh);
}

.forumTitle {
  display: flex;
  align-items: center;
  margin-bottom: 30px;
  font-weight: 400;
  font-size: 44px;
}

.form {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.textArea {
  width: 1200px;
  height: 400px;
  padding: 10px;
  border-radius: 12px;
  background-color: hsl(30deg 34% 84%);
  box-shadow: 2px 2px 5px 1px hsl(30deg 20% 20%) inset;
  outline: none;
  font-weight: 300;
  font-size: 40px;
  font-family: 'Rubik Bubbles', fantasy;
}

.button {
  width: 400px;
  margin-top: 20px;
  background-color: hsl(30deg 34% 84%);
  outline: none;
}
