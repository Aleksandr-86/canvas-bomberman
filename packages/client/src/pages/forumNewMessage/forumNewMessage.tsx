import { Button } from '../../components/button/button'
import { Link } from 'react-router-dom'
import classes from './forumNewMessage.module.css'
import { Input } from '../../components/input/input'

export const ForumNewMessage = () => {
  return (
    <div className={classes.forum}>
      <h2 className={classes.forum__title}>Новое сообщение</h2>

      <div className={classes.forum__wrapper}>
        <form>
          <Input type="text" placeholder="Введите сообщение" />
          <Button type="submit" className="forum__btn--create">
            сохранить
          </Button>
        </form>
        <Link to="/forum" className={classes.forum__link}>
          назад
        </Link>
      </div>
    </div>
  )
}
