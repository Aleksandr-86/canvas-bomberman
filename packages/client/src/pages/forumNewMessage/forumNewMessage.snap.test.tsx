import * as renderer from 'react-test-renderer'
import { ForumNewMessage } from './forumNewMessage'
import { MemoryRouter } from 'react-router-dom'

describe(`<ForumNewMessage/> snapshot test cases`, () => {
  test(`<ForumMessages/> should render <div className=forum> with child elements`, () => {
    const tree = renderer
      .create(
        <MemoryRouter initialEntries={[`/forum-new-message`]}>
          <ForumNewMessage />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
