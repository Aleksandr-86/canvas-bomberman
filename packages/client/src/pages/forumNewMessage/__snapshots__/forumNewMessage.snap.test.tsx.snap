// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForumNewMessage/> snapshot test cases <ForumMessages/> should render <div className=forum> with child elements 1`] = `
<div
  className="forum"
>
  <h2
    className="forumTitle"
  >
    Новое сообщение
  </h2>
  <div>
    <form
      onSubmit={[Function]}
    >
      <label>
        <input
          className="form__input"
          placeholder="Инпут"
          type="hidden"
        />
      </label>
      <label>
        <input
          className="form__input"
          name="body"
          placeholder="Введите сообщение"
          type="text"
        />
      </label>
      <button
        className="button forumBtnCreate"
        type="submit"
      >
        сохранить
      </button>
    </form>
    <a
      className="forumLink"
      href="/forum"
      onClick={[Function]}
    >
      назад
    </a>
  </div>
</div>
`;
